modules/serverpod_chat/serverpod_chat_server/lib/src/endpoints/chat_endpoint.dart

```diff
@@ -287,27 +300,48 @@ class ChatEndpoint extends Endpoint {
     int? thumbHeight;
 
     try {
-      const maxImageWidth = 256;
-
       var ext = path.extension(filePath.toLowerCase());
       if ({'.jpg', '.jpeg', '.png', '.gif'}.contains(ext)) {
         var response = await http.get(url!);
         var bytes = response.bodyBytes;
-        var image = decodeImage(bytes);
-        if (image != null) {
-          if (image.width > maxImageWidth) {
-            image = copyResize(image, width: maxImageWidth);
+        // Run thumbnail generation in an isolate, because it is CPU-intensive
+        var thumbnail = await Isolate.run(() {
+          var image = decodeImage(bytes);
+          if (image == null) {
+            return null;
           }
-          var thumbPath = _generateAttachmentFilePath(userId, fileName);
+          if (image.width > maxImageWidth || image.height > maxImageHeight) {
+            // Shrink image to fit thumbnail max size
+            int? width;
+            int? height;
+            if (image.width / maxImageWidth > image.height / maxImageHeight) {
+              width = maxImageWidth;
+            } else {
+              height = maxImageHeight;
+            }
```

Comment:
Discussion: The intention here was to always generate a cube thumbnail with sides of 256.

Feel free to ensure that always happens.
If we only pass one side to `copyResize` it will only resize based on that axel so we could just pass in the height and width always to the resize method.

Reply:
I'll leave this up to you, as I have no more time to work on this. I was just hoping to identify something that could really bog down Serverpod instances.

Reply:
Probably you should have the user pass in a `BoxFit` parameter, with `BoxFit.cover` as the default.

The other issue is alignment, if the source image is not square. The user should probably pass in a fraction from 0 (left/top) to 0.5 (center) to 1 (right/bottom), for the alignment fraction in the longest dimension, for if the source image is non-square. Default should be 0.5.

Reply:
Then we will focus on performance for now and just ensure that whatever thumbnail is outputted has the correct format.

The improvements you suggested are good and important enhancements, but we won't implement them at the moment..